operator/webhook/core/test/test_sync.py

```diff
@@ -189,6 +235,125 @@ def test_env_vars_no_keystore(full_route):
     assert not any(x for x in options if x.get("name") == "SERVER_SSL_KEYSTOREPASSWORD")
 
 
+def test_volume_pkcs12_keystore_and_pkcs12_truststore(full_route):
+    del full_route["parent"]["spec"]["tls"]["keystore"]
+    full_route["parent"]["spec"]["tls"]["keystore"] = {
+        "pkcs12": {
+            "secretName": "test-tls-secret",
+            "key": "test-keystore.p12",
+            "passwordSecretRef": "keystore-password-ref",
+        }
+    }
+    response = sync(full_route)
+    volumes = response["children"][0]["spec"]["template"]["spec"]["volumes"]
+    keystore_volume = None
+    truststore_volume = None
+    for volume in volumes:
+        if volume["name"] == "keystore":
+            keystore_volume = volume
+        elif volume["name"] == "truststore":
+            truststore_volume = volume
+
+    assert keystore_volume
+    assert keystore_volume["name"] == "keystore"
+    assert keystore_volume["secret"]["secretName"] == "test-tls-secret"
+    assert keystore_volume["secret"]["items"][0]["key"] == "test-keystore.p12"
+    assert keystore_volume["secret"]["items"][0]["path"] == "test-keystore.p12"
```

Comment:
:memo: nitpick, but might be simpler to build your expected `keystore_volume` dictionary and compare to actual in a single assert. Same for the other blocks below that follow the same pattern.

Reply:
Fixed